packages/block-editor/src/components/block-list/index.js

```diff
@@ -105,25 +105,19 @@ function Items( {
 				}
 				return newIntersectingBlocks;
 			} );
-		}
-
-		return new Observer( callback, { threshold: 0.1 } );
+		} );
 	}, [ setIntersectingBlocks ] );
-	const { order, selectedBlocks } = useSelect(
+	const { order, multiSelectedBlocks, selectedBlock } = useSelect(
 		( select ) => {
 			const {
 				getBlockOrder,
 				getSelectedBlockClientId,
 				getMultiSelectedBlockClientIds,
-				hasMultiSelection,
 			} = select( blockEditorStore );
 			return {
 				order: getBlockOrder( rootClientId ),
-				selectedBlocks: new Set(
-					hasMultiSelection()
-						? getMultiSelectedBlockClientIds()
-						: [ getSelectedBlockClientId() ]
-				),
+				multiSelectedBlocks: getMultiSelectedBlockClientIds(),
+				selectedBlock: getSelectedBlockClientId(),
```

Comment:
I think you're looking for getSelectedBlockClientIds (works for both multi and simple selection)

Reply:
Huh, I didn't know we had that. Strange that we weren't using it before here.